modules/backend/formwidgets/colorpicker/assets/js/colorpicker.js

```diff
@@ -57,24 +57,36 @@
                 chooseText: $.oc.lang.get('colorpicker.choose', 'Ok'),
                 cancelText: '⨯',
                 hide: function(color) {
-                    var hex = color ? color.toHexString() : ''
-                    self.$customColorSpan.css('background', hex)
+                    var customColor = self.covertColor(color)
+                    self.$customColorSpan.css('background', customColor)
                 },
                 show: function(color) {
                     self.selectColor(self.$customColor)
                 },
                 move: function(color) {
-                    var hex = color ? color.toHexString() : ''
-                    self.$customColorSpan.css('background', hex)
+                    var customColor = self.covertColor(color)
+                    self.$customColorSpan.css('background', customColor)
                 },
                 change: function(color) {
-                    var hex = color ? color.toHexString() : ''
-                    self.setCustomColor(hex)
+                    var customColor = self.covertColor(color)
+                    self.setCustomColor(customColor)
                 }
             })
         }
     }
 
+    /**
+     * Convert colour to hex or rgba depnding on alpha (opacity) value
+     */
+    ColorPicker.prototype.covertColor = function(color) {
+        if (color._a < 1 ) {
```

Comment:
If a color is undefined, it will throw an error.

Could you refactor this part? Start of by checking, if color is defined then alpha logic: `color._a < 1 ? color.toRgbString() : color.toHexString()`.

Reply:
Hi Denis,
1
I think verifying color looks a bit excessive. There should be something fundamentally wrong with 3rd-party Spectrum or Colorpicker partial, _colorpicker.htm, to get color undefined, etc. 1st is beyond the scope, 2nd looks ok so far. There is no verification for color in the original code as well.
 
2
Your shorthand code is perfect. I will teas and apply it. Thank you.